src/pudl_archiver/depositors/zenodo.py

```diff
@@ -0,0 +1,327 @@
+
+        Args:
+            deposition: the deposition you want to get the new version of.
+
+        Returns:
+            A new Deposition that is a snapshot of the old one you passed in,
+            with a new major version number.
+        """
+        if not deposition.submitted:
+            return deposition
+
+        url = f"{self.api_root}/deposit/depositions/{deposition.id_}/actions/newversion"
+
+        # Create the new version
+        headers = {
+            "Authorization": f"bearer {self.upload_key}",
+        }
+
+        # When the API creates a new version, it does not return the new one.
+        # It returns the old one with a link to the new one.
+        response = await self.request(
+            "POST", url, log_label="Creating new version", headers=headers
+        )
+        old_deposition = Deposition(**response)
+
+        source_metadata = old_deposition.metadata.dict(by_alias=True)
+        metadata = {}
+        for key, val in source_metadata.items():
+            if key not in ["doi", "prereserve_doi", "publication_date"]:
+                metadata[key] = val
+
+        previous = semantic_version.Version(source_metadata["version"])
+        version_info = previous.next_major()
+
+        metadata["version"] = str(version_info)
+
+        # Update metadata of new deposition with new version info
+        data = json.dumps({"metadata": metadata})
+
+        # Get url to newest deposition
+        new_deposition_url = old_deposition.links.latest_draft
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"bearer {self.upload_key}",
+        }
+
+        response = await self.request(
+            "PUT",
+            new_deposition_url,
+            log_label=f"Updating version number from {previous} ({old_deposition.id_}) to {version_info}",
+            data=data,
+            headers=headers,
+        )
+        return Deposition(**response)
+
+    async def publish_deposition(self, deposition: Deposition) -> Deposition:
+        """Publish a deposition.
+
+        Needs to have at least one file, and needs to not already be published.
+
+        Args:
+            deposition: the deposition you want to publish.
+        """
+        url = deposition.links.publish
+        headers = {
+            "Authorization": f"bearer {self.publish_key}",
+            "Content-Type": "application/json",
+        }
+        response = await self.request(
+            "POST", url, log_label="Publishing deposition", headers=headers
+        )
+        return Deposition(**response)
+
+    async def discard_deposition(self, deposition: Deposition) -> None:
+        """Discard a deposition.
+
+        This deposition must either be un-published or in the editing state.
+
+        Args:
+            deposition: the deposition you want to discard.
+
+        Returns:
+            None if success.
+        """
+        url = deposition.links.discard
```

Comment:
I implemented this seemingly very straightforward request, but immediately got a `500` for my trouble. Not sure if we should keep this around or not.